### PR TITLE
[multisite] Improve retrieval of individual‑site info in a multisite …

### PR DESCRIPTION
**Pull request by fajardoleo**
…network.

### CODE REVIEW

File: includes/class-freemius.php [hunk 1]
```diff
@@ -7897,18 +7897,20 @@ private function activate_bundle_license( $license, $sites = array(), $current_b
                         }
                     }
 
+                    $current_blog_id = get_current_blog_id();
```
Comment (swashata) [inline includes/class-freemius.php]: I am not fully aware of the mechanism, but don't we want to call this before even switching on line `7854`?

File: includes/class-fs-hook-snapshot.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+		private $removed_actions = array();
+
+		/**
+		 * Remove all actions from a given hook and store them for later restoration.
+		 */
+		public function remove( $hook ) {
+			global $wp_filter;
+
+			if ( isset( $wp_filter[ $hook ] ) ) {
```
Comment (DanieleAlessandra) [inline includes/class-fs-hook-snapshot.php]: In normal WP context this is good enough, but in edge cases it could be useful to safe check the existence of `$wp_filter`: 

```if ( isset( $wp_filter ) &&  isset( $wp_filter[ $hook ] ) ) {```

 This makes the code more robust in edge cases, such as when the file is loaded outside of the standard WordPress environment or during automated tests.
In normal WordPress usage, this is rarely necessary, but it helps prevent potential errors in atypical scenarios.

**Review by DanieleAlessandra — APPROVED** [review]
The changes look good enough to me, only read the one comment and think about it (we have to deal with tons of different use cases).

File: includes/class-freemius.php [hunk 3]
```diff
@@ -15832,9 +15833,8 @@ function get_site_info( $site = null, $load_registration = false ) {
                 $info[ 'registration_date' ] = $registration_date;
             }
 
-            if ( $switched ) {
```
Comment (swashata) [inline includes/class-freemius.php]: @fajardoleo why are we removing this? Don't we want to restore the blog like before?

**Review by swashata — COMMENTED** [review]
Please see my only question

**Review by swashata — APPROVED** [review]
Nice addition
